### ISSUE TITLE
GCAM customized policies

### ISSUE THREAD

**Issue by rp9823** [JGCRI/gcam-core#431]
I have 3 questions:
1. Is there any available guide on how to set up a customized policy on GCAM? 
2. If i want to run a scenario with one of the "pre - defined" policies, e.g. carbon_tax_10_5, should I add the policy to the scenario components in the configuration_ref file or in the configuration_policy? because I noticed they're slightly different.
3. If I set a scenario by fixing the emission pathway and solving for carbon price, should I enable the target finder mode?
Thanks

**Comment by Goldfish63** [comment]
Hi, Thanks for your questions. I also encountered similar issues you posted before and I would like to share my views.

1. The gcam-training offers some learning materials and some examples for you to grasp quickly. This tutorial provides some insights on how to design a policy scenario. Also, you can follow the examples on this website.
![image](https://github.com/JGCRI/gcam-core/assets/169158417/d8393211-dc5e-4df0-ae34-6cac5d575c64)

https://jgcri.github.io/gcam_training/gcam.html#Scenario_Adjustments

2. You should add the policy (e.g. carbon_tax_10_5) in the configuration_policy file. Configuration_ref file defines the reference scenario for GCAM release.

3. I think the find-path mode should be set to 1. This issue also mention the similar problem.https://github.com/JGCRI/gcam-core/issues/238

I am also quite new to gcam. Please correct if there are errors. Thanks again.

**Comment by pkyle** [comment]
Just to add to that, if I'm running a simple carbon price or emissions constraint scenario, I usually start from `configuration_ref.xml`, leave `find-path` set to 0, and add the carbon price or emissions constraint file to the `scenarioComponents`.
`find-path` set to 1 (as in `configuration_policy.xml`) indicates to use target-finder mode, which is used for solving a policy that will require multiple iterations of the model itself. For example, if you wanted to hit an end of century radiative forcing target, the exact concentrations of each gas in 2100 aren't known, as a number of different levels of each gas could work. Moreover,  the emissions pathways of each gas similarly are not known a priori. For such a policy design, the model will have to run a number of different times, adjusting the GHG price trajectory such that the end of century target is met. In this mode, only the final scenario run is saved and written to the output database.

**Comment by rp9823** [comment]
Thanks a lot for your help! I'd like to ask you another question. In the queries related to prices and costs, I found different labels, for example regional/total/traded/imported/delivered and so on. Is there a guide explaining in detail how to interpret the different prices/costs according to the labels? Furthermore, as far as you know, is it possible to consider one of the prices as an acquisition cost and another as selling price so that the difference between the two may represent the profit for that market? Thanks again for your support.

**Comment by pkyle** [comment]
I don't think there's a guide to describing the interrelationships of the different commodities, but if you run the query “inputs by sector”, that will show all the various intersectoral linkages, at least within a region. In general, a commodity prefaced by “regional" indicates the total domestic supply; this is equal to the sum of all consumption within the region, or equivalently, the production minus net exports. In general, gcam does not represent stock changes or statistical differences.
A commodity prefaced by “traded” tracks the quantity of the commodity that is traded between model regions. Traded commodities are housed in the USA region (though this is arbitrary; it could be any region), and the subsectors represent exporting regions. The traded commodities are the inputs to the corresponding “imports” subsectors of each regions “regional” commodities.
The trade structure in gcam is relatively simple; it does not consider bilateral trade, and all importing regions see the same price for a given imported commodity.
Past the trade structure, the commodity names are simpler; “delivered” or “enduse” generally refer to commodities delivered to end use consumers, often with cost markups that are consistent with a distributed network of enduse consumers. In contrast, “wholesale” or “industrial” Indicate the commodity price is paid by large, industrial or energy sector consumers.

**Comment by rp9823** [comment]
Thanks a lot for your answer. If you don't mind it, I have another question about GCAM input data. I noticed that final calibration year is 2015. Therefore I guess that the time series, used as inputs, stop at 2015. And this is true even for the last GCAM release which is the 7.1. My question is: Are there any available sets of input data updated to, say, 2020 or 2023, so to be able to extend the GCAM calibration period and run simulations with more recent data?

**Comment by pkyle** [comment]
There is an ongoing effort to update the base year, and to streamline the process of updating the base year so that future updates will be easier, but it's still under development. Among other challenges, in GCAM there is only one calibration year; we cannot have, for example, a calibration year of 2022 for energy and a calibration year of 2019 for agriculture. So, every time series data set used for calibrating the model needs to be updated to whatever the new calibration year is, or methods need to be devised for extrapolating forward datasets that are more lagged.

**Comment by rp9823** [comment]
Hi @pkyle, is there any news about the possibility to get updated input calibration data in the new GCAM releases?

**Comment by pkyle** [comment]
No. I believe we are aiming to update the model calibration year in the next model release.
